### PR TITLE
initial 1.89 ver

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
libansicon 1.89
 - [upstream](https://github.com/adoxa/ansicon)
 - windows only
 - dependency for ansicon

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c')}}
+    - m2-patch  # [win]
+
+about:
+  home: https://github.com/adoxa/ansicon
+  license: OTHER
+  license_family: OTHER
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
  license: Zlib
  license_family: Zlib
```
https://spdx.org/licenses/Zlib.html

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c')}}
+    - m2-patch  # [win]
+
+test:
+  commands:
+    - if not exist %PREFIX_LIB%\include\ansicon.h exit 1     # [win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: I don't think `PREFIX_LIB` is set in tests...

File: recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - if not exist %PREFIX_LIB%\include\ansicon.rc exit 1     # [win]
+    - if not exist %PREFIX_BIN%\ansicon.exe exit 1     # [win]
+    - if not exist %PREFIX_BIN%\ANSI64.dll exit 1     # [win]
+    - if not exist %PREFIX_BIN%\ANSI64.dll exit 1     # [win]
+    - if not exist %PREFIX_LIB%\ANSI64.exp exit 1     # [win]
+    - if not exist %PREFIX_LIB%\ANSI64.lib exit 1     # [win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - if not exist %LIBRARY_INC%\ansicon.h exit 1     # [win]
    - if not exist %LIBRARY_INC%\ansicon.rc exit 1     # [win]
    - if not exist %LIBRARY_BIN%\ansicon.exe exit 1     # [win]
    - if not exist %LIBRARY_BIN%\ANSI64.dll exit 1     # [win]
    - if not exist %LIBRARY_LIB%\ANSI64.exp exit 1     # [win]
    - if not exist %LIBRARY_LIB%\ANSI64.lib exit 1     # [win]
```

**Review by skupr-anaconda — APPROVED** [review]

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Good job @M-Waszkiewicz-Anaconda! This wasn't an easy one!

File: recipe/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/adoxa/ansicon/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: e8135c9f3e6c1249f9e6bbabc6996ac42c2853db922e3b067e96e0cd31596ab3
+  patches:
+    - 0001-fix-the-build-for-VS-2015.patch
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:12]:
```suggestion
  patches:  # [win]
    - 0001-fix-the-build-for-VS-2015.patch  # [win]
```

**Review by skupr-anaconda — COMMENTED** [review]
As it's only for windows

**Pull request merged**